app/components/templates/templates.html

```diff
@@ -195,6 +195,30 @@
               </div>
             </div>
             <!-- !volume-mapping -->
+            <!-- extra-host -->
+            <div class="form-group" >
+              <div class="col-sm-12" style="margin-top: 5px;">
+                <label class="control-label text-left">Hosts file entries</label>
+                <span class="label label-default interactive" style="margin-left: 10px;" ng-click="addExtraHost()">
+                  <i class="fa fa-plus-circle" aria-hidden="true"></i> add additional entry
+                </span>
+              </div>
+              <!-- extra-host-input-list -->
+              <div class="col-sm-12">
+                <div class="col-sm-12 form-inline" style="margin-top: 10px;">
+                  <div ng-repeat="host in state.selectedTemplate.Hosts" style="margin-top: 2px;">
```

Comment:
If we want the values to be correctly bound to the model, we need to use the following snippet here:

```html
<div ng-repeat="(idx, host) in state.selectedTemplate.Hosts" style="margin-top: 2px;">
    <div class="input-group col-sm-5 input-group-sm">
      <span class="input-group-addon">value</span>
      <input type="text" class="form-control" ng-model="state.selectedTemplate.Hosts[idx]" placeholder="e.g. host:IP">
    </div>
    <button class="btn btn-sm btn-danger" type="button" ng-click="removeExtraHost($index)">
      <i class="fa fa-trash" aria-hidden="true"></i>
    </button>
</div>
```

We should not bind to primitives, see: https://github.com/angular/angular.js/issues/1267